### PR TITLE
Fix wrong usage of kubelet option

### PR DESCRIPTION
**Pull request by bart0sh**
**What this PR does / why we need it**:

"--allow-privileged true" is incorrect usage of boolean option.
It means setting '--allow-priviledged' to its default value plus
non-existing subcommand 'true'.

"--allow-privileged false" is even more confusing as it sets
allow-priviledged flag to its default value 'true'

This is true for any boolean command line option.

Fixed this by using correct syntax --allow-priviledged=true

**Special notes for your reviewer**:
This is a show-stopper for PR #61833 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by feiskyer — COMMENTED** [review]
/lgtm

File: test/e2e_node/services/kubelet.go [hunk 1]
```diff
@@ -264,7 +264,7 @@ func (e *E2EServices) startKubelet() (*server, error) {
 		"--kubeconfig", kubeconfigPath,
 		"--root-dir", KubeletRootDirectory,
 		"--v", LOG_VERBOSITY_LEVEL, "--logtostderr",
-		"--allow-privileged", "true",
+		"--allow-privileged=true",
```
Comment (mtaufen) [inline test/e2e_node/services/kubelet.go]: we should also make sure boolean flags are handled properly in `addKubeletConfigFlags`, below
https://github.com/kubernetes/kubernetes/blob/5fa4f2f6ec7669c7f9ee6e290bc91501e8fd965b/test/e2e_node/services/kubelet.go#L352-L359

Comment (bart0sh) [reply test/e2e_node/services/kubelet.go]: Thank you for pointing this out. Will fix by this patch and push again:
```
--- a/test/e2e_node/services/kubelet.go
+++ b/test/e2e_node/services/kubelet.go
@@ -354,7 +354,7 @@ func addKubeletConfigFlags(cmdArgs *[]string, kc *kubeletconfig.KubeletConfigura
        fs := pflag.NewFlagSet("kubelet", pflag.ExitOnError)
        options.AddKubeletConfigFlags(fs, kc)
        for _, name := range flags {
-               *cmdArgs = append(*cmdArgs, "--"+name, fs.Lookup(name).Value.String())
+               *cmdArgs = append(*cmdArgs, fmt.Sprintf("--%s=%s", name, fs.Lookup(name).Value.String()))
        }
 }
```

**Comment by bart0sh** [comment]
kindly ping @Random-Liu  for a final review

**Review by dims — APPROVED** [review]

**Comment by bart0sh** [comment]
kindly ping  @feiskyer @mtaufen @Random-Liu for a final review/approval

**Pull request merged**